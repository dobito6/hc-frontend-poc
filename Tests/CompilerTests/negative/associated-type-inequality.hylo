//! no-std

trait P { type X }

fun check<T>(_ x: T) {}

fun f0<T, U where T == U, T is P, U is P>(x: U.X) {
  // We cannot conclude that `T.X` is equal to `U.X` here because the function takes different
  // conformances. Hence, even if `T` is equal to `U`, it is possible for their conformances to
  // have different associated types.
  check<T.X>(x)
}
