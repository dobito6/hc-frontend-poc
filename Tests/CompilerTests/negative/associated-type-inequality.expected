negative/associated-type-inequality.hylo:11.14-15: error: cannot pass value of type '(U::P).X' to parameter '(T::P).X'
  check<T.X>(x)
             ^
